### PR TITLE
feat(do-dont): adds new components DoDontRow, Do, Dont with built in grid

### LINKED ISSUE #497: [Spacing] Vertical spacing between image clusters should be 32px
Spacing in between images should be consistent both horizontally and vertically
https://www.carbondesignsystem.com/guidelines/accessibility/overview

Screen Shot 2019-11-05 at 4 43 47 PM

It should behave more like this:
https://gatsby-theme-carbon.now.sh/components/ImageGallery

Screen Shot 2019-11-05 at 4 48 20 PM

FYI 
@vpicone 
@mjabbink

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #497 
- For review: look at the _new_ DoDontRow page in the nav.

- Original issue requires custom row vertical spacing to match the gutter spacing. Also, per https://github.com/carbon-design-system/gatsby-theme-carbon/pull/554#pullrequestreview-325251971, adds built in columns in new `Do` / `Dont` components. 

- PR introduces **3 brand new components** so it's not a breaking change for those currently using `DoDontExample`. But rather people can migrate to these components until we deprecate the old component. 

- Same code structure/ styles, but refactored to use css modules.

- _Note: The `DoDontRow` acts as just a **single** row, not a wrapper for all examples being show (i.e. not like `ImageGallery`). Reasoning is that this way individuals can decide how many columns wide the row should be. Most do/dont examples span 2 columns, but sometimes (like in this page: https://www.ibm.com/design/language/typography/type-basics) they span 3 columns. Making the DoDontRow a row that can accommodate for all this would require supporting whitespace/offset columns in some way for different breakpoints. Any thoughts on this???_

Screen Shot 2020-01-10 at 2 35 56 PM
Screen Shot 2020-01-10 at 2 30 57 PM

#### Changelog

**New**

- `DoDontRow`
- `Do`
- `Dont`

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Although this is not a breaking change design/Mike is going to expect that all sites using the theme get updated to use this. Can we add migration docs and make sure once this is released its communicated on how/why to update the components. 

Also just wanted to be sure we couldn't achieve this without creating all new components? Updating all DoDont components seems like it will be a large undertaking, did a search on the IDL website and its used over 250 times, and over 100 on the Carbon website.

**Comment by vpicone** [comment]
@alisonjoseph Been a while since I looked at this but in order to control the spacing around the components (similar to the image gallery) they'd need a wrapper (the DoDontRow) which is a breaking change anyhow.

@jnm2377 I think we'd want one component that renders the markup/applies the essential classes then `Do` and `Dont` are just simple wrappers that apply the appropriate prop. That way the implementation is consistent between the two.

**Comment by jnm2377** [comment]
@vpicone wanna pair on this tomorrow during mob programming time?

**Comment by vpicone** [comment]
@jnm2377 totally. It looks clean as hell btw, great job!

**Comment by vpicone** [comment]
related maybe #568

**Comment by vpicone** [comment]
Looks great! Wonder if we should add a deprecation warning to the `DoDontExample` for a v2.

File: packages/gatsby-theme-carbon/src/components/DoDontRow/DoDontRow.module.scss [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  right: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+//---------------------------------------
+// Images
+//---------------------------------------
+.example-content .gatsby-resp-image-link {
```
[the remark this answers is not in the document]
Comment (jnm2377) [reply packages/gatsby-theme-carbon/src/components/DoDontRow/DoDontRow.module.scss]: @vpicone the first row example has an image. is that what you mean?

**Comment by jnm2377** [comment]
@sadek this PR would resolve the spacing issue with do-dont example clusters. Is this visually correct?

File: packages/gatsby-theme-carbon/src/components/DoDontRow/DoDontRow.module.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpicone) [reply packages/gatsby-theme-carbon/src/components/DoDontRow/DoDontRow.module.scss]: @jnm2377 yeah, perfect. I like the way you documented the column props!

**Comment by sadekbazaraa** [comment]
Yup, this looks right to me @jnm2377

**Comment by vpicone** [comment]
Wooo! Gonna do a feature release momentarily.

**Pull request merged**